MetaMask/metamask-extension
# Fix bundle size diff message

The bundle size diff message is using the wrong point of comparision, leading to misleading results on feature branches that have been merged with `develop` since they were created.

When this feature was introduced, we went back and forth a few times on what we should be comparing the branch with to get an accurate bundle size comparison.

The first attempt used `develop` as the point of comparison, but that didn't work because it was a moving target, and because it didn't reflect the changes made on this branch. As bundle increases or decreases were added on `develop`, they would alter the diff on each feature PR.

Then we chose to use the fork-point of the branch, the commit of `develop` that the branch forked off of. This works for feature branches that don't merge in `develop`. But the minute `develop` gets merged in, then unrelated changes on `develop` affect the measurement.

The _most recent_ commit from `develop` on the current branch is a better comparison. Any difference between this commit and the feature branch in terms of bundle size would be attributable to the feature branch changes. This is what `merge-base` gives us.

## Manual Testing Steps

The diff message should show zero changes.

It's hard to test the branch cases directly because this always uses `develop` as the point of comparison, so we'll have to just watch future PRs to ensure the results make sense.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
